### PR TITLE
Include PATCH as an option to perform HTTP requests

### PR DESCRIPTION
**Pull request by mandicaBrito**
Includes PATCH as an option to perform HTTP requests:

- it adds PATCH as HTTP option

- it updates related request classes

- it includes unit tests

__Dependencies:__
- [x] Unit Tests

### CODE REVIEW

File: api-client/src/test/java/com/xing/api/CallSpecTest.java [hunk 1]
```diff
@@ -399,6 +399,48 @@ public void builderAllowsJsonBody() throws Exception {
         assertRequestHasBody(callSpec.request(), expectedBody, 24);
     }
 
+    @Test
```
Comment (juliagarrigos) [inline api-client/src/test/java/com/xing/api/CallSpecTest.java]: I'm unsure about if we need these tests, the logic they are testing, is already tested right? Do we need to re-test it again for the different methods? GET, POST, PATCH?

File: gradle.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 GROUP_ID=com.xing.api
-VERSION=3.1.0
-VERSION_CODE=17
+VERSION=3.1.1
```
Comment (juliagarrigos) [inline gradle.properties:2]: Also not sure about when we should increase the version number, @MyDogTom can we do it in this PR and create a release when it is closed?

File: api-client/src/test/java/com/xing/api/CallSpecTest.java (hunk 1 above)
Comment (mandicaBrito) [reply api-client/src/test/java/com/xing/api/CallSpecTest.java]: We don't need to specifically check for each HTTP method, but all the others are being tested, so I added tests for PATCH too. I can remove them.

File: gradle.properties (hunk 2 above)
Comment (MyDogTom) [reply gradle.properties:2]: Since we don't have any other changes planed, let's increase version right here.

**Review by juliagarrigos — APPROVED** [review]

**Pull request merged**